### PR TITLE
Refactor handleCommand to return specific state type.

### PR DESCRIPTION
**Pull request by wketting**
Should we remove NoState and make state Optional in AggegateRoot state?
We could add a DefaultAggregateRoot class with default (AggregateState) state handling. Or even dynamically construct state (props/map/etc.).

### CODE REVIEW

**Comment by olger** [comment]
Having Option of aggregate root state is a good idea. NoState can be removed as that is None. 
The currentState in the TestableAggregateRoot should also return the specific state. (should be added)

I'd prefer not to have a default with default state handling for now. (but maybe I don't understand what it brings the user of the framework)

File: bounded-core/src/main/scala/io/cafienne/bounded/aggregate/AggregateRootActor.scala [hunk 1]
```diff
@@ -35,10 +35,17 @@ trait AggregateRootActor extends PersistentActor with AggregateStateCreator with
...
     *
     * @param command one of the commands your aggregate root expects
-    * @param state of the Aggregate Root at the moment of handling command
+    * @param state of the Aggregate Root at the moment of handling command, optional: state could not be set yet
     * @return a sequence of events when everything is Right, or an Failure (Left)
     */
-  def handleCommand(command: DomainCommand, state: AggregateState): Reply
+  def handleCommand(command: DomainCommand, state: Option[A]): Reply
+
+  private def handleCommand(command: DomainCommand, state: AggregateState): Reply = {
```
Comment (olger) [inline bounded-core/src/main/scala/io/cafienne/bounded/aggregate/AggregateRootActor.scala]: this private method should not be necessary ? it is called by receiveCommand and that can call it without transformation by passing the Option of the state

Comment (olger) [reply bounded-core/src/main/scala/io/cafienne/bounded/aggregate/AggregateRootActor.scala]: the ```def state: Option[AggregateState] ``` van be typed to A in order to ensure that the state is strongly typed

Comment (olger) [reply bounded-core/src/main/scala/io/cafienne/bounded/aggregate/AggregateRootActor.scala]: And the AggregateStateCreator should return the specific typed state.

File: bounded-test/src/main/scala/io/cafienne/bounded/test/TestableAggregateRoot.scala [hunk 2]
```diff
@@ -44,7 +44,7 @@ object TestableAggregateRoot {
     * @tparam A The Aggregate Root Type that is tested.
     * @return a TestableAggregateRoot instance that is initialized and available to give a DomainCommand to.
     */
-  def given[A <: AggregateRootActor](
+  def given[A <: AggregateRootActor[_]](
```
Comment (olger) [inline bounded-test/src/main/scala/io/cafienne/bounded/test/TestableAggregateRoot.scala]: AggregateRootActor[B <: AggregateState] ?

File: bounded-core/src/main/scala/io/cafienne/bounded/aggregate/AggregateRootActor.scala [hunk 3]
```diff
@@ -79,13 +86,12 @@ trait AggregateRootActor extends PersistentActor with AggregateStateCreator with
...
     case _: RecoveryCompleted                               => // initialize further processing when required
-    case SnapshotOffer(_, snapshot: Option[AggregateState]) => internalState = snapshot
+    case SnapshotOffer(_, snapshot: Option[A]) => internalState = snapshot
     case evt: DomainEvent                                   => updateState(evt)
 
     case other =>
       log.warning("Received unknown event {} during recovery", other)
   }
-
 }
```
Comment (olger) [inline bounded-core/src/main/scala/io/cafienne/bounded/aggregate/AggregateRootActor.scala]: NoState can be removed that will solve some ambiguity in the code

File: cargo-sample/src/main/scala/io/cafienne/bounded/cargosample/domain/Cargo.scala [hunk 4]
```diff
@@ -16,13 +16,11 @@ import scala.collection.immutable.Seq
...
 class Cargo(cargoId: AggregateRootId, locationsProvider: LocationsProvider)
-    extends AggregateRootActor
-    with AggregateStateCreator
-    with ActorLogging {
+  extends AggregateRootActor[CargoAggregateState] {
 
   override def aggregateId: AggregateRootId = cargoId
 
-  override def handleCommand(command: DomainCommand, currentState: AggregateState): Reply = {
+  override def handleCommand(command: DomainCommand, state: Option[CargoAggregateState]): Reply = {
```
Comment (olger) [inline cargo-sample/src/main/scala/io/cafienne/bounded/cargosample/domain/Cargo.scala]: :)

**Comment by wketting** [comment]
@olger Adjustments made as requested. Plz review.

**Pull request merged**